src/BarChart/BarSeries/BarTargetMarker.tsx

```diff
@@ -0,0 +1,216 @@
+import React, { FC, useMemo } from 'react';
+import { motion } from 'framer-motion';
+import {
+  ChartInternalShallowDataShape,
+  DEFAULT_TRANSITION,
+  Direction
+} from '@/common';
+
+export interface BarTargetMarkerProps {
+  /**
+   * Height of the bar.
+   */
+  height: number;
+
+  /**
+   * Width of the bar.
+   */
+  width: number;
+
+  /**
+   * SVG x attribute for the bar.
+   */
+  x: number;
+
+  /**
+   * SVG y attribute for the bar.
+   */
+  y: number;
+
+  /**
+   * Group index or index of the bar. Set internally by `BarSeries`.
+   */
+  index: number;
+
+  /**
+   * D3 scale for Axis. Set internally by `BarChart`.
+   */
+  scale: any;
+
+  /**
+   * Parsed data shape. Set internally by `BarChart`.
+   */
+  data: ChartInternalShallowDataShape;
+
+  /**
+   * Color of the target marker line.
+   */
+  fill: string;
+
+  /**
+   * Color of the positive delta line.
+   */
+  positiveDeltaFill: string;
+
+  /**
+   * Color of the negative delta line.
+   */
+  negativeDeltaFill: string;
+
+  /**
+   * Number of the bars in the bar group. Set internally by `BarSeries`.
+   */
+  barCount: number;
+
+  /**
+   * Layout of bar chart to render. Set internally by `BarSeries`.
+   */
+  layout: Direction;
+
+  /**
+   * Whether to animate the enter/update/exit. Set internally by `BarSeries`.
+   */
+  animated: boolean;
+
+  /**
+   * Class name to apply to the text.
+   */
+  className?: string;
+
+  /**
+   * Thickness of the target marker line.
+   */
+  targetLineThickness?: number;
```

Comment:
Try to follow standard naming patterns `targetStrokeWidth` - https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-width